app/javascript/components/subject/GenericAction.js

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Form, Button, Modal } from 'react-bootstrap';
+import _ from 'lodash';
+import axios from 'axios';
+
+import InfoTooltip from '../util/InfoTooltip';
+import reportError from '../util/ReportError';
+
+class GenericAction extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      patient: props.patient,
+      showExposureRiskAssessmentModal: false,
```

Comment:
So I think we have two options with this component:

1. Break it out into separate components completely because it's not really generic enough to justify putting it all in one.
2. Make this component even more generic so that it can be used for almost all of the monitoring actions and doesn't contain so much action-specific code. We might still want there to be a new component for every action, but there would be much less code in each since the skeleton component would contain a good chunk of it. 

I'd much prefer #2 if we think it's do-able. Here's what I'm thinking:
- Every component in the Monitoring Actions is a dropdown with a modal. Make this a "dumb" (presentational) generic component `MonitoringStatusDropdown` that just has that skeleton (modal with checkboxes, change handler(s), dropdown). 
- Everything else should be handled by props. You can pass routes as props, you can pass the handlers as props if need be as well. We may still want wrapper components around these to pass the necessary props. 
Basically taking advantage of [this pattern](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0).
Thoughts? I just don't think this as it is is super useful - we're running into the same problem as before where it's too much in one.

Reply:
My thoughts on that is I feel like #2 would become too much like we had before so it might make sense to just do individualized ones...  I know neither is preferable but I do feel like with #1 it would be MUCH easier to follow and definitely cleaner.  Let me know which you're thinking and I can change it.  Going to make the other changes for now

Reply:
I think we can get the clarity of #1 with the removal of duplication of #2. 

Here's what I'm thinking, let me know how this sounds. I'll use `AssignedUser` as an example.
- New component `MonitoringActionsDropdown` that contains the following:
    - Contains you already have in `AssignedUser.js` (createModal, submit, render, toggleModal, handleKeyPress) EXCEPT you make it not specific to AssignedUser, and you accept the POST route, handlers, and state change callback(s) as props. 
- Update component `AssignedUser.js` that  maintains the current non-presentational state logic (i.e. `apply_to_group`, `assigned_user`, `reasoning`, etc.) that it's already maintaining, but instantiates the `MonitoringActionsDropdown` with the correct props, and this component manages presentation logic such as `loading` and `showModal`.

 This way `AssignedUser` is still separate and still handles all the logic, but `MonitoringActionsDropdown` handles all the repeated modal, dropdown, and submission logic. Kind of like a new util or layout component. It'd be similar to the `CustomTable` presentational component that's being used by the `PatientsTable` and the `AdminTable`

Do you foresee any issues with this that I may be overlooking?

Reply:
The issue that I think might be the big one is then we're getting back to the point where the createModal is going to be a nightmare of logic and html (like it was before).  I get not wanting to repeat code but I just can't see having a generic createModal that would be better than before...

Maybe I'm not fully understanding but I don't think that would be really much better than we had before, perhaps less confusing by a bit.

The other thing is that Jurisdiction and Assigned User are different than the other actions and would therefore still have to be separate on a rendering level

Reply:
Thinking about this some more... a good amount of the Case/Monitoring Status modal logic is handled by state change and is more complicated.  Factor that in with the fact that the jurisdiction/assigned user inputs are different then you're looking at Monitoring Plan, Public Health Action and Exposure Risk Assessment which I originally broke out into that generic copmponent.

Long answer but I'm not sure there's a good answer here.  If we keep it too general, it may start looking similar to what it was before, but we definitely are repeating code by separating them.  this would be good though if some of these become more custom than they already are (like how case status got)

Reply:
Well the idea would be that that logic handling would still be done in the `CaseStatus` component (for example), and the generic component would just handle displaying the modal and calling the correct callbacks which would be in props.`createModal` shouldn't have to deal with any logic. 

I do think this would be do-able without getting logic mixed up because the generic component would be purely presentational, but I don't think it's a necessary part of this refactor for now. And if they do get more complex for any reason this does make it more straightforward, so I'm okay with just splitting it out here - which I think you already did, so I can resolve this thread.

Reply:
With all the cases the modal currently has I feel like that would be difficult - however once the PO team addresses how they want to handle everything moving forward that might get easier